test/dlc_tests/sanity/test_safety_check_v2.py

```diff
@@ -0,0 +1,97 @@
+import json
+import logging
+import sys
+
+import pytest
+
+from invoke import run
+from dataclasses import dataclass
+from typing import List
+
+from test.test_utils import (
+    CONTAINER_TESTS_PREFIX, is_dlc_cicd_context, is_canary_context, is_mainline_context, is_time_for_canary_safety_scan
+)
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+LOGGER.addHandler(logging.StreamHandler(sys.stderr))
+
+SAFETY_FILE = '/var/safety_report.json'
+SAFETY_FILE_EXISTS = 0
+
+
+@dataclass
+class SafetyVulnerabilityAdvisory:
+    vid: str
+    advisory: str
+    reason_to_ignore: str
+    spec: str
+
+
+@dataclass
+class SafetyPackageVulnerabilityReport:
+    package: str
+    scan_status: str
+    installed: str
+    vulnerabilities: List[SafetyVulnerabilityAdvisory]
+    date: str
+
+    def __post_init__(self):
+        self.vulnerabilities = [SafetyVulnerabilityAdvisory(**i) for i in self.vulnerabilities]
+
+
+@dataclass
+class SafetyPythonEnvironmentVulnerabilityReport:
+    report: List[SafetyPackageVulnerabilityReport]
+
+    def __post_init__(self):
+        self.report = [SafetyPackageVulnerabilityReport(**i) for i in self.report]
+
+
+
+@pytest.mark.model("N/A")
+@pytest.mark.canary("Run safety tests regularly on production images")
```

Comment:
Will the canary safety test override `/var/safety_report.json` before running this function?

Reply:
Thanks for pointing this out. 

The `test_safety_check_v2.py` will run only on PR and Mainline Context. On the Canaries, we will still run the old script that existed. This is done because the old images do not have a baked in safety report in them. Additionally, the purpose of the canaries is to run the safety scans on the images and generate fresh safety reports every day. 

I just have to make a small change to prevent `test_safety_check_v2.py` from running on canary and prevent the old script from running on PR and Mainline Context.